### PR TITLE
Add tests to recruiter's my jobs page

### PR DESCRIPTION
**Pull request by guyy13**
Added tests for the 'my jobs' page.
Changed the view to return a 404 error if the requesting user is unauthorized.

### CODE REVIEW

File: recruiter/tests.py [hunk 1]
```diff
@@ -147,3 +179,24 @@ def test_new_recruiter_account_with_valid_data(valid_recruiter_data, client):
...
+        list(jobs_and_applications_from_response[job]) == list(Application.get_applications_by_job(job)) for job in
+        jobs_and_applications_from_response)
+
+
+@pytest.mark.django_db
+def test_user_that_is_not_a_recruiter_cannot_access_my_jobs_url(example_student, client):
+    response = client.get("/myjobs")
+    assert response.status_code == 404
+    client.force_login(example_student.user)
+    assert response.status_code == 404
```
Comment (YarinBou) [inline recruiter/tests.py:210]: lgtm

**Review by YarinBou — APPROVED** [review]

**Review by yaelerdr — APPROVED** [review]
Looks good 👍

**Review by eldaraharonson — APPROVED** [review]

**Review by benmenachem — APPROVED** [review]
lgtm

File: recruiter/tests.py [hunk 2]
```diff
@@ -147,3 +179,24 @@ def test_new_recruiter_account_with_valid_data(valid_recruiter_data, client):
     assert example_recruiter_from_db.email == valid_recruiter_data["email"]
     assert example_recruiter_from_db.name == valid_recruiter_data["name"]
     assert example_recruiter_from_db.phone_number == valid_recruiter_data["phone_number"]
+
+
+@pytest.mark.django_db
+def test_recruiter_my_jobs_loads_correct_data(example_recruiter, example_job_application, client):
+    recruiter_object = Recruiter.objects.get(name="a")
+    client.force_login(recruiter_object.user)
+    response = client.get("/myjobs")
```
Comment (ifireball) [inline recruiter/tests.py:190]: Missing template check

File: recruiter/tests.py [hunk 3]
```diff
@@ -147,3 +179,24 @@ def test_new_recruiter_account_with_valid_data(valid_recruiter_data, client):
...
+    assert all(
+        list(jobs_and_applications_from_response[job]) == list(Application.get_applications_by_job(job)) for job in
+        jobs_and_applications_from_response)
+
+
+@pytest.mark.django_db
+def test_user_that_is_not_a_recruiter_cannot_access_my_jobs_url(example_student, client):
+    response = client.get("/myjobs")
+    assert response.status_code == 404
+    client.force_login(example_student.user)
```
Comment (ifireball) [inline recruiter/tests.py]: What is a point of doing a force_login without also doing an HTTP request? 
Please try to limit each test to a single HTTP call.

**Review by ifireball — CHANGES_REQUESTED** [review]

File: recruiter/tests.py (hunk 3 above)
Comment (guyy13) [reply recruiter/tests.py]: My bad I forgot to make another request.
I'll separate it into two tests.

File: recruiter/tests.py (hunk 2 above)
Comment (guyy13) [reply recruiter/tests.py:190]: On it

File: recruiter/tests.py (hunk 3 above)
Comment (guyy13) [reply recruiter/tests.py]: done

File: recruiter/tests.py (hunk 2 above)
Comment (guyy13) [reply recruiter/tests.py:190]: done

**Review by ifireball — APPROVED** [review]

**Pull request merged**